src/values/AbstractObjectValue.js

```diff
@@ -204,6 +204,58 @@ export default class AbstractObjectValue extends AbstractValue {
     return this;
   }
 
+  // ECMA262 9.1.1
+  $GetPrototypeOf(): ObjectValue | AbstractObjectValue | NullValue {
+    if (this.values.isTop()) {
+      AbstractValue.reportIntrospectionError(this);
+      throw new FatalError();
+    }
+    let elements = this.values.getElements();
+    if (elements.size === 1) {
+      for (let cv of elements) {
+        invariant(cv instanceof ObjectValue);
+        return cv.$GetPrototypeOf();
+      }
+      invariant(false);
+    } else if (this.kind === "conditional") {
```

Comment:
Annoying that one has to write so much code for what looks like a common and rather straightforward abstract value transformation... Could that be factored out?

Reply:
This one is pretty straightforward but the other places where this happens are not as straightforward. So I think we should probably wait and see what other patterns happen before we jump to abstractions.